src/txmempool.h

```diff
@@ -485,7 +485,13 @@ class CTxMemPool
         >
     > indexed_transaction_set;
 
-    mutable CCriticalSection cs;
+    /**
+     * Mutex that guards reads.
```

Comment:
In commit "doc: Add comment to cs_main and mempool::cs" (fa64bb42e97b6c6c58e83badca88d386f99dbcd2)

I think making a comparison to read/write locks makes this more confusing than it needs to be. Read locks normally are shared, not exclusive. And the requirement to lock `cs_main` for write access seems too broad, and doesn't give a justification.

I would suggest a comment that just says when you need to lock the mutexes and why:

```c++
/**
 * This mutex needs to be locked when accessing mapTx.
 *
 * @par Relationship to cs_main
 * Locking both mempool.cs and cs_main gives a view of the mempool which is
 * compatible with the current chain tip. To guarantee this, both mutexes need
 * to be locked when adding transactions to the mempool or changing the
 * chain tip. But locking mempool.cs by itself is sufficient for reading
 * transactions from the mempool, or removing transactions from it.
 */
```

Reply:
It could also be used to guard other members, like prioritization data, no?

Reply:
Otherwise done